src/Protocol/ActivityPub/Processor.php

```diff
@@ -268,6 +247,62 @@ public static function createEvent($activity, $item)
 		Logger::log('Event '.$event_id.' was stored', Logger::DEBUG);
 	}
 
+	/**
+	 * Process the content
+	 *
+	 * @param array $activity Activity array
+	 * @param array $item
+	 * @return array
+	 * @throws \Exception
+	 */
+	private static function processContent($activity, $item)
+	{
+		$item['title'] = HTML::toBBCode($activity['name']);
+
+		if (!empty($activity['source'])) {
+			$item['body'] = $activity['source'];
+		} else {
+			$content = HTML::toBBCode($activity['content']);
+			$content = self::replaceEmojis($content, $activity['emojis']);
+			$content = self::convertMentions($content);
+
+			if (($item['thr-parent'] != $item['uri']) && ($item['gravity'] == GRAVITY_COMMENT)) {
+				$item_private = !in_array(0, $activity['item_receiver']);
+				$parent = Item::selectFirst(['id', 'private', 'author-link', 'alias'], ['uri' => $item['thr-parent']]);
+				if (!DBA::isResult($parent)) {
+					Logger::warning('Unknown parent item.', ['uri' => $item['thr-parent']]);
+					return false;
```

Comment:
Please return an empty array (`[]`) instead (no type mismatching). And AFAIK `empty` is working with empty arrays too.

Reply:
I thought of this like - for example - the ```strpos``` function in PHP that can return ```0``` when the found content is at position zero, and that does return ```false``` when nothing is found at all.

I don't see a problem in mixed types here.

Reply:
Ah, OK. I just read about PHP 7 and that they introduced typehint for specific return types. So it seems for me that - on a long run - they want to streamline mixed return types to specific ones. That's why I requested these change :-).

Can you add something like `@return array|bool Returns the item array or false if there was an unexpected occurrence` to the phpdoc?

Reply:
done

Reply:
I'm with @nupplaphil on this one, `strpos` indeed needs two different return value types because like you said, `0` is a success value despite being falsy.

However, in this case, an empty array isn't a success value, which makes it a good candidate for an error return value and allow to have only one return value type.